wordpress-mobile/WordPress-Android
# Make the secondary toolbar the main toolbar in the Reader post list view

**Fixes #7847** 

![screen shot 2018-06-06 at 11 41 54 am](https://user-images.githubusercontent.com/5810477/41070520-857de96e-69b0-11e8-9e0a-a0bc23f7856c.png)


This PR turns the secondary toolbar on the Reader view into the main toolbar. Since toolbar uses a custom view with a spinner for filtering items I couldn't reuse the toolbar in `WPMainActivity`. To get around this, I removed the toolbar in `WPMainActivity` and moved the toolbar to the main fragments hosted by `WPMainNavigationView`. The fragments are:
- `MeFragment`
- `MySiteFragment`
- `ReaderPostListFragment`
- `NotificationsListFragment`

Each of the above fragments implements a new `ToolbarFragment` interface to allow Activity to set the title on each of the fragment's toolbar (when applicable). 

All but `ReaderPostListFragment` share the same toolbar layout `toolbar_main.xml`. Since each fragment hosts its own toolbar and `FilteredRecyclerView` includes a packaged toolbar, it made sense to retain as much of the original layout for this fragment to avoid unnecessary code duplication and management. To make `FilteredRecyclerView` more flexible for use as a primary toolbar, I added some custom styleable attributes:
- `wpSpinnerItemView` - Optional: layout resource id to override the default
- `wpSpinnerDropDownItemView` - Optional: layout resource id to override the default view 
- `wpToolbarVisibilityLock` - Optional: Default false. If true, this will prevent the scroll to hide/reveal the toolbar, as well as the ability to swipe up to hide the toolbar.

The fade transition used when switching between these 4 main fragments has been removed. When enabled, there was a slight but still noticeable flash in the toolbar when switching fragments since the toolbar belongs to the fragment it is subject to the same fade transition. 

**To test:**
1. Open **Reader** view - verify there is only now 1 toolbar and its styling matches the toolbars in the other 3 main fragments.
2. Scroll on the reader tab - verify the toolbar stays visible.
3. Open **My Sites** view and click on **Stats** - verify the secondary toolbar is there and when you scroll it hides/reveals itself. 
   - This view also uses `FilteredRecyclerView` so it's a good sanity check that the new styleable attributes are defaulting properly.
4. Flip through each bottom bar option and verify toolbar loads as expected. 
6. Switch sites, verify title is set to the appropriate site on when on the **My Site** view.
7. Open **Notifications** view - verify scrolling properly hides the secondary toolbar.
